File: services/tapo-service/internal/service/models/device.go
package models

type DeviceConnectionInfo struct {
	DeviceId string `json:"device_id"` // The same ID as the TapoDevice

	DeviceType string `json:"device_type"`
	IpAddress  string `json:"ip_address"`
	Email      string `json:"email"`
	Password   string `json:"password"`
}

type TapoDevice struct {
	DeviceId   string                 `json:"device_id"`
	DeviceName string                 `json:"device_name"`
	DeviceType string                 `json:"device_type"`
	DeviceInfo map[string]interface{} `json:"device_info"`
}

File: services/tapo-service_deprecated/registry/device.py
from typing import Dict


class TapoDevice:
    def __init__(self, ip_address: str, email: str, password: str, device_type: str):
        self.ip_address = ip_address
        self.email = email
        self.password = password
        self.device_type = device_type
        self.device_name = ""
        # self.initialized = False

    def turn_on(self):
        pass

    def turn_off(self):
        pass

    def get_device_info(self) -> Dict[str, str]:
        pass

    def get_device_name(self) -> str:
        pass

    def get_device_type(self) -> str:
        pass

    def set_device_name(self, name: str):
        pass

    def wake_up(self) -> bool:
        pass
